examples/errorcall_back_agent.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+import logging
+import os
+
+from dotenv import load_dotenv
+
+from livekit.agents import JobContext, WorkerOptions, cli
+from livekit.agents.metrics import LLMMetrics
+from livekit.agents.voice import Agent, AgentSession, MetricsCollectedEvent
+from livekit.plugins import cartesia, deepgram, openai, silero
+
+logger = logging.getLogger("my-worker")
+logger.setLevel(logging.INFO)
+
+load_dotenv()
+
+
+class MyTask(Agent):
+    def __init__(self):
+        super().__init__(
+            instructions="You are a helpful assistant that can answer questions and help with tasks.",  # noqa: E501
+        )
+
+
+async def entrypoint(ctx: JobContext):
+    await ctx.connect()
+
+    session = AgentSession(
+        stt=deepgram.STT(),
+        llm=openai.LLM(),
+        tts=cartesia.TTS(),
+        vad=silero.VAD.load(),
+    )
+
+    close_task = None
+
+    @session.on("metrics_collected")
+    def on_metrics_collected(ev: MetricsCollectedEvent):
+        nonlocal close_task
+        if isinstance(ev.metrics, LLMMetrics) and ev.metrics.error:
+            if not ev.metrics.error.retryable or ev.metrics.error.attempts_remaining == 0:
+                logger.info("Ran into an unrecoverable LLM error, ending session.")
+                # do something with the error
+                close_task = asyncio.create_task(session.aclose())
+                close_task.add_done_callback(lambda _: logger.info("Session closed"))
+        return
+
+    await session.start(agent=MyTask(), room=ctx.room)
+
+
+if __name__ == "__main__":
+    # set the OPENAI_API_KEY to a invalid key to simulate an error
+    os.environ["OPENAI_API_KEY"] = "invalid"
```

Comment:
For the example, we should find a better example to simulate failures, maybe editing the llm_node